### DISCUSSION TITLE
Defer helper is not working

### DISCUSSION THREAD

**Discussion by sanjaykm098** [laravel/framework#52904] [Q&A] [answered]
Hi. In my Laravel app v-11.23.5 the defer is not working. My app migrated from version 10.23 to version 11 2 months ago.

**Comment by crissi** [comment]
Same for me, the closure is never called

**Reply by Aurabolt** [reply]
Same for me, I can't get defer to work. I'm on 11.23.5

**Comment by mahmoud-almalah** [comment]


**Reply by Aurabolt** [reply]
This doesn't work for me either. And I'm now on 11.25

**Reply by henzeb** [reply]
Do you have fast-cgi enabled?

**Reply by Aurabolt** [reply]
@henzeb  yeah. php-fpm 8.2

**Comment by lotestudio** [comment]
Laravel 11 does not use Kernel, instead, it uses application bootstrap to define and register middleware. For deffer to work you must add \Illuminate\Foundation\Http\Middleware\InvokeDeferredCallbacks::class middleware in your web middlewareGroup in Http\Kernel.php

```
protected $middlewareGroups = [
        'web' => [
            \Illuminate\Foundation\Http\Middleware\InvokeDeferredCallbacks::class,
            \App\Http\Middleware\EncryptCookies::class,
            \Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse::class,
           ...

**Reply by MoussaPro** [reply, marked as answer]
Worked, thanks!

**Reply by Aurabolt** [reply]
Thanks @lotestudio , that worked for me too! I also found the section of Laravel documentation hat we all seem to have missed that confirms your solution:

https://laravel.com/docs/11.x/helpers#deferred-function-compatibility

**Reply by lotestudio** [reply]
This is missing in the upgrade prologue  for 10 to 11 version. Maybe it should be there too...

**Reply by lamka02sk** [reply]
Thank you, this should be in the docs.
